src/main/java/org/beanpod/switchboard/service/EncoderServiceImpl.java

```diff
@@ -12,20 +13,23 @@
 
 @Component
 @RequiredArgsConstructor
+@Slf4j
 public class EncoderServiceImpl implements EncoderService {
   private final EncoderDaoImpl encoderDao;
   private final StreamDaoImpl streamDao;
   private final DateUtil dateUtil;
 
   @Override
   public List<StreamDto> getEncoderStreams(String encoderSerialNumber) {
+    log.info("Getting Encoder's {} streams", encoderSerialNumber);
     EncoderDto encoderDto =
         encoderDao
             .findEncoder(encoderSerialNumber)
             .orElseThrow(() -> new ExceptionType.DeviceNotFoundException(encoderSerialNumber));
 
     encoderDto.setLastCommunication(dateUtil.getCurrentDate());
     encoderDao.save(encoderDto);
+    log.debug("Updated encoder's last communication date");
```

Comment:
This log should also include the encoder serial number since, due to asynchronous execution, it might not appear right after the previous one so we won't know which encoder got updated. Same for decoder.

Reply:
fixed